docs/modding/squirrel/what-is-squirrel.md

```diff
@@ -0,0 +1,59 @@
+What is Squirrel
+==========
+Squirrel is a programming language used in titanfall 2 and northstar mods, it is worth noting that the version of squirrel used by titanfall appears to be a modified version of the langauge
+
+File format
+----------
+
+All files using squirrel are saved as .nut or .gnut
+
+when creating mods your mod will typically be a series of functions and callbacks
+
+Squirrel as a typed language
+========
+All variables and functions in squirrel must have a type defined on declaration
+
+```cpp
+function dosomething()
+```
+
+is not acceptable and should instead be
+
+```cpp
+void function dosomething()
+```
+Similarly when declaring a variable
+```cpp
+funnynumber = 69
+```
+will not work, instead use:
+```cpp
+int funnynumber = 69
+```
+
+
+Variable types
+-------------
+
+`void`: can be used to define functions that do not return a value, but still do things. most of your functions will be void
+
+```interger```: a whole number
+
+```float```: a none-whole number.
+
+```string```: text
+
+```entity```: an entity and all its associated information
+
+```bool```: true or false (Squirrel does not treat 1 as true or 0 as false)
+
+```array```: an ordered list of items indexed from 0, increasing with the number of items such as: 0:"hello",1:1000,2:entity player ...
+
+```table```: a list of items with indexes defined on object entry such as: "word":"hello", "kilo":1000", "pilot":entity player ...
+
+```struct```: a series of attributes assigned to a structure and accessed through with structname.variablename
+
+```var```: i hate these, they can be an output however they cannot be used as an input for any functions.

```

Comment:
It is worth noting that `array` and `table` contain `var` data by default. You can strictly type them with `<type>` for arrays and `<key_type, value_type>` for tables. For instance, `array<int> numList` forces all values added to `numList` to be ints and also lets you treat its values as ints. I.e.
```cpp
array<int> numList = [0, 1]
int myNum = numList[0] 
// This works as expected

array list = [0, 1]
int myNum2 = list[0]
// Breaks; list's elements are vars, and you cannot assign a var to an int (or any other static type) without an expect

array<int> badList = [0, "hello"]
int myNum3 = badList[0]
// Breaks; badList can only contain ints

```
On another note, `var` can totally be used as input. There is nothing that restricts their usage outside of an attempt to assign them to a strictly typed variable. For that, you have to convert them via `expect your_type( yourVar )`, though sometimes you may get yelled at by the Client if it expects `yourVar` to be `your_type`. Though, I do recall there being some things you need an `untyped` file to do, but I can't remember what they are at this time (it isn't using var as input though, that works normally).

Reply:
finished the dedicated page for tables and arrays, improved the description for var